Add empty plugins dir for archive distributions

jasontedor opened the pull request:
Today when installing Elasticsearch from an archive distribution (tar.gz
or zip), an empty plugins folder is not included. This means that if you
install Elasticsearch and immediately run elasticsearch-plugin list, you
will receive an error message about the plugins directory missing. While
the plugins directory would be created when starting Elasticsearch for
the first time, it would be better to just include an empty plugins
directory in the archive distributions. This commit makes this the
case. Note that the package distributions already include an empty
plugins folder.

Closes #20342

jasontedor committed:
Add empty plugins dir for archive distributions

Today when installing Elasticsearch from an archive distribution (tar.gz
or zip), an empty plugins folder is not included. This means that if you
install Elasticsearch and immediately run elasticsearch-plugin list, you
will receive an error message about the plugins directory missing. While
the plugins directory would be created when starting Elasticsearch for
the first time, it would be better to just include an empty plugins
directory in the archive distributions. This commit makes this the
case. Note that the package distributions already include an empty
plugins folder.

nik9000 reviewed (approved):
Looks right. Can you add a test that runs the plugin command without any plugins installed?

jasontedor committed:
Add test that listing plugins works after install

This commit adds a test that listing plugins does not fail with an
exception after installing Elasticsearch but before running
Elasticsearch for the first time.

jasontedor commented:
@nik9000 I pushed c411dd6d6806aa0d4a01c40e199882c5c34d194f.

jasontedor commented:
test this please

jasontedor commented:
@nik9000 Are you good with the test that was pushed in c411dd6d6806aa0d4a01c40e199882c5c34d194f.

nik9000 commented on qa/vagrant/src/test/resources/packaging/scripts/20_tar_package.bats:
```diff
@@ -73,6 +73,13 @@ setup() {
     verify_archive_installation
 }
 
+@test "[TAR] verify elasticsearch-plugin list runs" {
```
s/runs/runs without any plugins installed/

nik9000 commented:
Tests look good I'd change the name of the test case so it is super obvious.

jasontedor committed:
Change name of plugin list test

This commit changes the name of the plugin list packaging test so that
the intent is obvious.

jasontedor commented:
> Tests look good I'd change the name of the test case so it is super obvious.

Sure, I pushed e42513c1a0e4521929e33a019068979ad5dadda0. Note that the test does assert that the output from `elasticsearch-plugin list` is empty to ensure that the test is not moved somewhere after some plugins are installed.
